Plugins/Encounter/Encounter.cpp

```diff
@@ -281,94 +281,73 @@ ArgumentStack Encounter::GetCurrentNumSpawned(ArgumentStack&& args)
-        {
-            retVal.reserve(32 * pEncounter->m_nNumActivateVertices);
+        retVal.reserve(32 * pEncounter->m_nNumActivateVertices);
 
-            for(int i = 0; i < pEncounter->m_nNumActivateVertices; i++)
-            {
-                retVal += "{" + std::to_string(pEncounter->m_pvActivateVertices[i].x) + ", " +
-                                std::to_string(pEncounter->m_pvActivateVertices[i].y) + ", " +
-                                std::to_string(pEncounter->m_pvActivateVertices[i].z) + "}";
-            }
-        }
-        else
+        for(int i = 0; i < pEncounter->m_nNumActivateVertices; i++)
         {
-            LOG_WARNING("NWNX_Encounter_GetEncounterGeometry() called on non encounter object.");
+            retVal += "{" + std::to_string(pEncounter->m_pvActivateVertices[i].x) + ", " +
+                            std::to_string(pEncounter->m_pvActivateVertices[i].y) + ", " +
+                            std::to_string(pEncounter->m_pvActivateVertices[i].z) + "}";
         }
     }
-
     return Services::Events::Arguments(retVal);
 }
 
-ArgumentStack Encounter::SetEncounterGeometry(ArgumentStack&& args)
+ArgumentStack Encounter::SetGeometry(ArgumentStack&& args)
 {
-    if (auto *pObject = encounter(args))
+    if (auto *pEncounter = encounter(args))
     {
         const auto sGeometry = Services::Events::ExtractArgument<std::string>(args);
 
-        if (auto *pEncounter = Utils::AsNWSEncounter(pObject))
-        {
-            auto str = sGeometry.c_str();
-            std::vector<Vector> vecVerts;
-            Vector vec = {};
-
-            do {
-                int cnt = std::sscanf(str, "{%f,%f,%f", &vec.x, &vec.y, &vec.z);
-
-                if (cnt != 2 && cnt != 3)
-                {
-                    LOG_NOTICE("NWNX_Encounter_SetEncounterGeometry() invalid geometry string at: %s", str);
-                    break;
-                }
-
-                if (cnt == 2)
-                    vec.z = pEncounter->GetArea()->ComputeHeight(vec);
+        auto str = sGeometry.c_str();
+        std::vector<Vector> vecVerts;
+        Vector vec = {};
 
-                vecVerts.push_back(vec);
-            } while ((str = std::strstr(str + 1, "{")));
+        do {
+            int cnt = std::sscanf(str, "{%f,%f,%f", &vec.x, &vec.y, &vec.z);
 
-            if (vecVerts.size() > 2)
+            if (cnt != 2 && cnt != 3)
             {
-                CNWSArea *pArea = pEncounter->GetArea();
-                pEncounter->RemoveFromArea();
+                LOG_NOTICE("NWNX_Encounter_SetGeometry() invalid geometry string at: %s", str);
+                break;
+            }
+
+            if (cnt == 2)
+                vec.z = pEncounter->GetArea()->ComputeHeight(vec);
 
-                if (pEncounter->m_pvActivateVertices)
-                    delete[] pEncounter->m_pvActivateVertices;
-                if (pEncounter->m_pnOutlineVertices)
-                    delete[] pEncounter->m_pnOutlineVertices;
```

Comment:
Only the if statements can be removed, but both `delete[] ... ` lines have to stay

Reply:
My fault for not being clear. I edited it.